nntrainer/layers/layer_context.h

```diff
@@ -94,12 +95,16 @@ class InitLayerContext {
    * @param reg_const regularization constant for the weight
    * @param name name of the weight
    * @param trainable if the weight is trainable (require gradient or not)
+   * @param opt_vars_dim vector of optimizer variable dimension
    * @return unsigned int index of the weight for its getter
    */
-  unsigned int requestWeight(const TensorDim &dim, const WeightInitializer init,
-                             const WeightRegularizer reg, const float reg_const,
-                             std::string name, bool trainable = true) {
-    weights_spec.emplace_back(dim, init, reg, reg_const, trainable, name);
+  unsigned int requestWeight(
+    const TensorDim &dim, const WeightInitializer init,
+    const WeightRegularizer reg, const float reg_const, std::string name,
+    bool trainable = true,
+    std::vector<TensorDim> opt_vars_dim = std::vector<TensorDim>()) {
```

Comment:
`requestWeight` must not have opt_vars_dim for weight in its interface. It cannot be decided by the user, but only by the optimizer.

Reply:
Hmm... That means I need to remove opt_vars_dim in interface. I will amend the commit based on your opinion